### PR TITLE
HPCC-22313 Strip path from <Filename>filename</Filename> in <Exception> line of test result.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add check to enable strip the path from <Exception> line.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Tested locally and manually with "HPCC-22196 Ensure text of failure is available to FAILMESSAGE #12642" code.

### CODE REVIEW

**Comment by richardkchapman** [comment]
I'm not convinced this is the right approach. We should not be generating fully-expanded paths in our exceptions (or in our graphs).

@ghalliday What do you think?

**Comment by AttilaVamos** [comment]
@ghalliday @richardkchapman this PR is ready to decision.

(BTW: If I see well in assert.ecl, there is already implemented something similar : "#option ('reportAssertFilenameTail', true);" )

**Comment by ghalliday** [comment]
@richardkchapman I had a look at fixing this in the code generator.  It was relatively simple to fix the paths found via -I to only include the relevant trailing path.  However it isn't clear what to do with the filename that was supplied to compile the query.  If it is within the include paths then it could be mapped, but what if it isn't?

**Comment by ghalliday** [comment]
The other problem would be that filenames reported from the code generator would no longer be absolute pathnames - which would make it hard for ecl ide to resolve the correct file.

File: testing/regress/ecl/key/assert.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 <Exception><Code>100000</Code><Filename>assert.ecl</Filename><Line>46</Line><Source>user</Source><Message>Assert (2 = 1) failed [val1 = 1]</Message></Exception>
 <Exception><Code>100000</Code><Filename>assert.ecl</Filename><Line>47</Line><Source>user</Source><Message>Abc7</Message></Exception>
 <Exception><Code>100000</Code><Filename>assert.ecl</Filename><Line>67</Line><Source>user</Source><Message>Assert (2 = 9) failed [COUNT(ds) = 9]</Message></Exception>
-<Exception><Code>100000</Code><Filename>assert.ecl</Filename><Line>71</Line><Source>user</Source><Message>Assert (&#226;&#130;&#172; = &#226;&#130;&#172;&#226;&#130;&#172;) failed [euro = U&apos;&#226;&#130;&#172;&#226;&#130;&#172;&apos;]</Message></Exception>
+<Exception><Code>100000</Code><Filename>assert.ecl</Filename><Line>71</Line><Source>user</Source><Message>Assert (&#226;&#130;&#172; = &#226;&#130;&#172;&#226;&#130;&#172;) failed [euro = U'&#226;&#130;&#172;&#226;&#130;&#172;']</Message></Exception>
```
Comment (richardkchapman) [inline testing/regress/ecl/key/assert.xml]: This change seems unrelated
